Shared/alerts/AlertConditionsController.cpp

```diff
@@ -710,6 +710,58 @@ void AlertConditionsController::onLayersChanged()
   addStoredConditions();
 }
 
+const std::optional<qint64> AlertConditionsController::getPickedElementId(const QList<GeoElement*>& geoElements) const
+{
+  // check again after the completion of the identify and abort if tool
+  // is no longer the active tool or there were no geoelements returned
+  if (!isActive())
+    return {};
+  if (geoElements.count() < 1)
+    return {};
+
+  // get the entity id for dynamic entity types
+  auto* geoElement = geoElements.first();
+  if (auto* observation = dynamic_cast<DynamicEntityObservation*>(geoElement); observation)
+  {
+    observation->deleteLater();
+    return observation->dynamicEntity()->entityId();
+  }
+
+  // check the attribute table if feature type
+  if (auto* feature = dynamic_cast<Feature*>(geoElement); feature)
+  {
+    const auto* attributes = feature->attributes();
+    auto* table = feature->featureTable();
+    if (!attributes)
+      return {};
+    if (!table)
+      return {};
+
+    const auto primaryKeyField = primaryKeyFieldName(table);
+    if (primaryKeyField.isEmpty())
+      return {};
+    if (!attributes->containsAttribute(primaryKeyField))
+      return {};
+
+    return attributes->attributeValue(primaryKeyField).toInt();
```

Comment:
On all code paths where you return an actual value, I think `std::make_optional` or just `std::optional{ ... }` is more explicit. My two cents would be to add something like that to make it clearer to the reader that's what is really happening. Without it, it is an implicit conversion.